aws/timeout_read_closer.go

```diff
@@ -56,3 +60,44 @@ func (r *timeoutReadCloser) Read(b []byte) (int, error) {
 func (r *timeoutReadCloser) Close() error {
 	return r.reader.Close()
 }
+
+// AddResponseReadTimeoutMiddleware adds a middleware to the stack that wraps the
+// response body so that a read that takes too long will return an error.
+func AddResponseReadTimeoutMiddleware(stack *middleware.Stack, duration time.Duration) {
+	stack.Deserialize.Add(&readTimeoutMiddleware{duration: duration}, middleware.After)
+}
```

Comment:
This middleware probably should go in the SDK's `aws/transport/http` package since it is specific to `smithyhttp.Request`.  Ideally the `aws` package is transport agnostic.